src/compas_blender/artists/frameartist.py

```diff
@@ -89,13 +91,13 @@ def draw_axes(self) -> List[bpy.types.Object]:
         -------
         list of :class:`bpy.types.Object`
         """
-        origin = list(self.frame.point)
-        X = list(self.frame.point + self.frame.xaxis.scaled(self.scale))
-        Y = list(self.frame.point + self.frame.yaxis.scaled(self.scale))
-        Z = list(self.frame.point + self.frame.zaxis.scaled(self.scale))
+        origin = list(self.primitive.point)
+        X = list(self.primitive.point + self.primitive.xaxis.scaled(self.scale))
+        Y = list(self.primitive.point + self.primitive.yaxis.scaled(self.scale))
+        Z = list(self.primitive.point + self.primitive.zaxis.scaled(self.scale))
         lines = [
-            {'start': origin, 'end': X, 'color': self.color_xaxis, 'name': f"{self.frame.name}.xaxis"},
-            {'start': origin, 'end': Y, 'color': self.color_yaxis, 'name': f"{self.frame.name}.yaxis"},
-            {'start': origin, 'end': Z, 'color': self.color_zaxis, 'name': f"{self.frame.name}.zaxis"},
+            {'start': origin, 'end': X, 'color': self.color_xaxis, 'name': f"{self.primitive.name}.xaxis"},
+            {'start': origin, 'end': Y, 'color': self.color_yaxis, 'name': f"{self.primitive.name}.yaxis"},
+            {'start': origin, 'end': Z, 'color': self.color_zaxis, 'name': f"{self.primitive.name}.zaxis"},
```

Comment:
somewhat of a "note to self"... perhaps it would make sense to start using named tuples for this. they have a much lower memory footprint and fit the purpose of these mini dicts better...